From 038f5e221ae58ca8a5111ff263722a0f70f79316 Mon Sep 17 00:00:00 2001
From: Lynxtron Scripts <scripts@lynxtron>
Date: Mon, 7 Sep 2026 19:23:37 +0800
Subject: [PATCH] Build perfetto for HarmonyOS

Provide the vm_sockets definitions the OHOS NDK sysroot does not ship.
---
 src/base/vm_sockets.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/base/vm_sockets.h b/src/base/vm_sockets.h
index ad59b5d9..d7349ee9 100644
--- a/src/base/vm_sockets.h
+++ b/src/base/vm_sockets.h
@@ -27,7 +27,7 @@
 #if PERFETTO_BUILDFLAG(PERFETTO_OS_QNX)
 // Requires the QNX Advanced Virtualization Framework
 #include <vm_sockets.h>
-#elif defined(AF_VSOCK)
+#elif defined(AF_VSOCK) && !defined(__OHOS__)
 // Use system vm_socket.h if available.
 #include <linux/vm_sockets.h>
 #else  // defined(AF_SOCK)
